extensions/amp-ad/0.1/test/test-amp-ad-3p-impl.js

```diff
@@ -179,10 +180,10 @@ describe('amp-ad-3p-impl', () => {
   });
 
   describe('preconnectCallback', () => {
-    it('should add preconnect and prefech to DOM header', done => {
+    it('should add preconnect and prefech to DOM header', () => {
       ad3p.buildCallback();
       ad3p.preconnectCallback();
-      setTimeout(() => {
+      return whenFirstVisible.then(() => {
```

Comment:
Could be 1 of two things:

- [Assertion](https://github.com/ampproject/amphtml/blob/28dcf17331237772eceb822797c8a46b96c72f12/extensions/amp-ad/0.1/test/test-amp-ad-3p-impl.js#L190-L194) is being [swallowed](https://github.com/ampproject/amphtml/issues/8966) (because this [callback](https://github.com/ampproject/amphtml/blob/28dcf17331237772eceb822797c8a46b96c72f12/src/preconnect.js#L200-L214) has not fired yet)
- `setTimeout` has been throttled